src/main/java/cys4/model/ExtensionEntity.java

```diff
@@ -11,23 +11,31 @@
 
 public class ExtensionEntity {
     private boolean active;
-    private String extension;
-    private transient Pattern extension_regex_compiled;
+    private final String extension;
+    private final transient Pattern extension_regex_compiled;
     private final String description;
 
-    public ExtensionEntity(String description, String extension) {
+    public ExtensionEntity(String description, String extension) throws IllegalArgumentException {
         this(description, extension, true);
     }
 
-    public ExtensionEntity(String description, String extension, boolean active) {
+    public ExtensionEntity(String description, String extension, boolean active) throws IllegalArgumentException {
+        if (extension == null || extension.isBlank())
+            throw new IllegalArgumentException("Invalid regex");
+
         this.active = active;
-        this.extension = extension;
         this.description = description;
-        this.extension_regex_compiled = null;
 
-        if (this.extension != null && !this.extension.isBlank() && !this.extension.endsWith("$")) {
-            this.extension += '$';
+        if (extension.endsWith("$")) {
+            this.extension = extension;
+        } else {
+            this.extension = extension + '$';
         }
+        this.extension_regex_compiled = Pattern.compile(this.extension);
```

Comment:
[PatternSyntaxException](https://docs.oracle.com/javase/7/docs/api/java/util/regex/PatternSyntaxException.html)

Reply:
IllegalArgumentException is already a parent of PatternSyntaxException